x-pack/plugins/global_search/server/routes/find.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { reduce, map } from 'rxjs/operators';
+import { schema } from '@kbn/config-schema';
+import { IRouter } from 'src/core/server';
+
+export const registerInternalFindRoute = (router: IRouter) => {
+  router.post(
+    {
+      path: '/internal/global_search/find',
+      validate: {
+        body: schema.object({
+          term: schema.string(),
+          options: schema.maybe(
+            schema.object({
+              preference: schema.maybe(schema.string()),
+            })
+          ),
+        }),
+      },
+    },
+    async (ctx, req, res) => {
+      const { term, options } = req.body;
+      try {
+        const allResults = await ctx
+          .globalSearch!.find(term, { ...options, aborted$: req.events.aborted$ })
+          .pipe(
+            map((batch) => batch.results),
+            reduce((acc, results) => [...acc, ...results])
+          )
+          .toPromise();
+        return res.ok({
+          body: {
+            results: allResults,
+          },
+        });
+      } catch (e) {
+        return res.forbidden({ body: e });
```

Comment:
This seems like a bad default response for all exceptions. Can we only return a forbidden response in the cases where we know that is the problem?

Reply:
added in https://github.com/elastic/kibana/pull/66293/commits/fa3b8c0207e4d5d5f4d828f883dad5c7a77938f4